### PR TITLE
Match Cleanup

### PR DESCRIPTION
**Pull request by avatarneil**
@thenewwazoo Mind taking a look at this? Happy to explain more of my changes if you'd like!
- Cargo Fmt to clean up a couple things.
- Changed unnecessary matches into maps.
- Added cases for locked/unknown to FlashError rather than a default. 
  - Previously, the default would handle _any_ cases without cargo throwing an error, possibly permitting future additions to the FlashError enum to not be handled explicitly and fly under the radar.
  - Now, if any more enums are added to FlashError, cargo will complain that they need to be handled, versus just filing them under `_`.  Ex (note, I did implement handling for Locked and Unknown!): 
```error[E0004]: non-exhaustive patterns: `&Locked` and `&Unknown` not covered
  --> src/flash.rs:95:15
   |
95 |         match self {
   |               ^^^^ patterns `&Locked` and `&Unknown` not covered```

### CODE REVIEW

File: src/flash.rs [hunk 1]
```diff
@@ -93,13 +93,14 @@ impl FlashError {
...
-            FlashError::NotZero => sr.inner().modify(|_,w| w.notzeroerr().set_bit()),
-            FlashError::FetchAbort => sr.inner().modify(|_,w| w.fwwerr().set_bit()),
-            _ => {},
+            FlashError::WriteProtect => sr.inner().modify(|_, w| w.wrperr().set_bit()),
+            FlashError::Alignment => sr.inner().modify(|_, w| w.pgaerr().set_bit()),
+            FlashError::Size => sr.inner().modify(|_, w| w.sizerr().set_bit()),
+            FlashError::ReadProtection => sr.inner().modify(|_, w| w.rderr().set_bit()),
+            FlashError::NotZero => sr.inner().modify(|_, w| w.notzeroerr().set_bit()),
+            FlashError::FetchAbort => sr.inner().modify(|_, w| w.fwwerr().set_bit()),
+            FlashError::Locked => ({}),
```
Comment (thenewwazoo) [inline src/flash.rs]: `FlashError::FetchAbort | FlashError::Locked => {},`

Comment (avatarneil) [reply src/flash.rs]: Did you mean `FlashError::Locked | FlashError::Unknown`? If so, done https://github.com/thenewwazoo/stm32l0x1-hal/pull/1/commits/626a746037b4136b593e15f1af230406aa46413b.

Comment (thenewwazoo) [reply src/flash.rs]: Yeah, sorry, was talking while I typed.

Comment (thenewwazoo) [reply src/flash.rs]: Also, `({})` is unnecessary. Just do `{}`

File: src/flash.rs [hunk 2]
```diff
@@ -374,18 +376,14 @@ impl WriteErase for Flash where Flash: Locking {
...
             ptr::write_volatile(address as *mut u32, 0);
         }
 
         while self.sr.is_busy() {}
 
-        let result = match self.status() {
-            Ok(_) => Ok(()),
-            Err(e) => Err(e),
-        };
+        let result = self.status().map(|_| ());
```
Comment (thenewwazoo) [inline src/flash.rs:386]: What is the return type of `status()`, and what happens when you map `_` to `()`? What is the resulting type?

Comment (thenewwazoo) [reply src/flash.rs:386]: Ah, okay, I see. lgtm. squash and I'll merge.

File: src/flash.rs (hunk 1 above)
Comment (avatarneil) [reply src/flash.rs]: If I just have `{}` then fmt gets rid of the trailing `,`. But I'm happy to make that change if you'd like!

Comment (thenewwazoo) [reply src/flash.rs]: Believe it or not, having that trailing `,` can cause weird issues with compiler output based on how it builds AST spans. Anyway, `rustfmt` knows all.

**Review by thenewwazoo — APPROVED** [review]

**Pull request closed without merging**